server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -268,6 +288,51 @@ class InteractiveSession(
     }
   }
 
+  private def datanucleusJars(livyConf: LivyConf): ArrayBuffer[String] = {
+    if (sys.env.getOrElse("LIVY_INTEGRATION_TEST", "false").toBoolean) {
+      // datanucleus jars has already been in classpath in integration test
+      ArrayBuffer.empty
+    } else {
+      val jars = ArrayBuffer.empty[String]
+      val sparkHome = livyConf.sparkHome().get
+      val libdir =
+        if (new File(sparkHome, "RELEASE").isFile) {
+          new File(sparkHome, "lib")
+        } else {
+          new File(sparkHome, "lib_managed/jars")
+        }
+
+      if (libdir.isDirectory()) {
+        jars ++= libdir.listFiles().filter(_.getName.startsWith("datanucleus-"))
+          .map(_.getAbsolutePath)
+      }
+      require(jars.nonEmpty, "datanucleus jars can not be found")
+      jars
+    }
+  }
+
+  /**
+   * Look for hive-site.xml
+   * 1. First look for under classpath
+   * 2. If not found under classpath, then look for that under SPARK_CONF_DIR and SPARK_HOME/conf
+   * @param livyConf
+   * @return
+   */
+  private def hiveSiteFile(livyConf: LivyConf): Option[File] = {
+    val hiveSiteURL = getClass.getResource("/hive-site.xml")
+    if (hiveSiteURL != null && hiveSiteURL.getProtocol == "file") {
+      return Some(new File(hiveSiteURL.toURI))
+    }
+    val hiveSiteFile = sys.env.get("SPARK_CONF_DIR")
```

Comment:
These checks are redundant; `bin/livy-server` already adds all these directories to the classpath, so the check above will already find the file.